### PR TITLE
Update Postgres to 14.10, 15.5, and 16.1

### PR DESCRIPTION
**Pull request by tristan957**
## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [x] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by lubennikovaav — CHANGES_REQUESTED** [review]
This PR is here just to run CI, right?
I don't see any open PRs in postrgres repo.


This releases fix a bug in indexes and mention that it is advisable to run REINDEX.
Does this issue affects neon databases too? I assume it does. Are we going to handle it somehow? At least we should communicate this properly in our Release notes.

**Comment by tristan957** [comment]
> This PR is here just to run CI, right?

Just spinning the CI.

**Comment by tristan957** [comment]
> This releases fix a bug in indexes and mention that it is advisable to run REINDEX.
Does this issue affects neon databases too? I assume it does. Are we going to handle it somehow? At least we should communicate this properly in our Release notes.

See the message @hlinnaka posted in #team-support on Slack for further discussion.

File: .gitmodules [hunk 1]
```diff
@@ -1,12 +1,12 @@
 [submodule "vendor/postgres-v14"]
 	path = vendor/postgres-v14
 	url = https://github.com/neondatabase/postgres.git
-	branch = REL_14_STABLE_neon
+	branch = tristan957/v14.10
```
Comment (bayandin) [inline .gitmodules:4]: Is it for testing purposes? Or just a wrong branch?

Comment (tristan957) [reply .gitmodules:4]: Testing purposes. See docs/updating-postgres.md

**Comment by bayandin** [comment]
The backward/forward tests have failed after restoring from WAL:

From pageserver [logs](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5835/6816986541/index.html#suites/0fbaeb107ef328d03993d44a1fb15690/4aa7f83d02933096?attachment=9fd291163617680b):
```
2023-11-10T16:27:04.864956Z ERROR wal_connection_manager{tenant_id=40d37eef472f1e208420eaac2c574744 timeline_id=e54b02cc9d6b738675acb0667e755ea8}: wal receiver task finished with an error: walreceiver connection handling failure

Caused by:
    invalid xl_tot_len 0 at 0/14EEA60
```

https://github.com/neondatabase/neon/issues/2592 most likely caused the failure.

**Comment by tristan957** [comment]
I think I need to do something with release notes? I am confused in what section I should write some notes.

**Comment by bayandin** [comment]
> I think I need to do something with release notes? I am confused in what section I should write some notes.

Just let @danieltprice know about it 😉

**Review by lubennikovaav — APPROVED** [review]
LGTM

**Pull request merged**